sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -157,6 +157,7 @@ public static DataSourceConfiguration create(
       checkNotNull(url, "url");
       checkNotNull(username, "username");
       checkNotNull(password, "password");
+      checkArgument(password.length() > 0, "password");
```

Comment:
* Change message to "password must be non-empty" - the exception will otherwise not give any information indicating that, it'll simply say something like "IllegalArgumentException: password".
* Why must the password be non-empty? Aren't there cases where the password for a username is allowed to be empty?
* Must the username, url, etc be non-empty as well, or are they allowed to be empty? (I'm not sure it's responsibility of this class to verify these - I'd think we should let the JDBC driver verify them according to this particular database's rules).